### ISSUE TITLE
Suggestion: search bar for "Favorites" category

### ISSUE THREAD

**Issue by Devonek** [PedroS11/twitch-live-extension#53]
First of all, love the extension. Thanks for your work.

Is there any possibility you'll be adding _search_ function for the "Favorites" category? It would save a lot of scrolling and searching,

**Comment by PedroS11** [comment]
Thank you so much for it, I appreciate it!!
Adding the search function can be done but there's an issue with it, so when you open the extension, it loads 50 followed streams that are live, the search function would only be able to search on those 50 and not all your followed because Twitch API doesn't allow it. If you have more than 50 and you scroll to the bottom it would load 50 more and so on. 
Basically, the search would only be able to filter streams that are already loaded on the extension. Do you think that's something useful despite this issue?

**Comment by Devonek** [comment]
Oh, I thought you had access to all live channels at once. I still think it could be useful, but I see the issue it might bring, mainly confusion. Personally, I never have more than 50 channels live at the same time, but this could be a problem for someone else.

I have an idea, but I don't know the technical limitations. Here is the idea for an algorithm:
1. Search through the 50 live channels.
2. If there is a match, display it.
3. If there is no match, load 50 more channels and try again (repeat until there are no more live channels to load).

This idea might not be feasible, but it would be great if it could work. The last extension I used before yours had a search bar, so it never occurred to me that it could be more complicated than that. However, they might have changed how the API works since then.

**Comment by PedroS11** [comment]
Yeah, loading all the followed streams in memory might cause memory issues for users that follow a lot of people, that's why I have it 50 at each time.
The older extensions used a now deprecated Twitch Api that I can no longer use and that limited everyone. I think I can try another way but for it to work, as you're my sample on this let me know :D If you input the **exact** twitch username, I think I can check if you follow it and if it's live, who it be something useful? It has to be the exact one, if you input just part of it, I think I won't be able to get it

**Comment by Devonek** [comment]
Alright, got it. Using the exact username is probably not the best way to go, since I usually don't remember every character in their name, and I suppose many other people don't either. So, let's just leave it as it is. Thanks for the explanation and effort. The extension might be easier to use without the exact-match search bar. If you agree, I think we can close the thread. Cheers!

**Comment by PedroS11** [comment]
I asked around on Twitch API discord and it might be "possible" using multiple calls to their API. Basically use [this](https://dev.twitch.tv/docs/api/reference/#search-channels) to get streams by input (doesn't need to be exact just like in the search bar) and then check if from the list of results which ones you follow using this [endpoint](https://dev.twitch.tv/docs/api/reference/#get-users-follows). My main concern is performance but it's something that I can evaluate while coding too. So on the Favorites, you'd be able to search by name the Live and Followed streams. Would this fit your use case?

**Comment by PedroS11** [comment]
This feature is now live as I just pushed a new version with it. Let me know if you find any issues

**Comment by Devonek** [comment]
Aw man I forgot to answer last time, sorry about that. I just downloaded the update and it seems to be working like a charm! Exactly what I had in mind. You're the man, thank you so much. I'll let you know if something breaks in the future, but I don't see any issues right this time.

**Comment by PedroS11** [comment]
No worries!! I'll keep this open for a couple of days, if you find anything just let me know. I tested with my own account and everything seemed fine but I may have way fewer follows than the average twitch user

**Comment by Devonek** [comment]
I took the time to follow some streamers, so I now have more than 50 streams live at once (52 to be exact), and everything worked smoothly. The search function worked as expected, so it seems that all is good to go.

I have one suggestion that might make the extension more user-friendly for lazy people like myself: could the search bar be active as soon as the extension loads, so users don't have to click into it themselves? This is just a small suggestion, so don't worry if it doesn't make sense to implement. I thought that since scrolling works even when the search bar is active, it might make the search process faster for those who use it, without bothering anyone else who doesn't use the search bar.

**Comment by PedroS11** [comment]
That's good news, thank you!!

I can have a look at it and see how to implement it, I don't think it should be that complicated, if it isn't, I don't mind adding it, that's something that didn't cross my mind. but nice catch

**Comment by PedroS11** [comment]
I was a super easy change, just pushed it, during today, you should see it working, let me know if there's any issue!
